src/pages/smart-contracts/clarity-values-guide.md

```diff
@@ -0,0 +1,143 @@
+
+## Clarity Types
+The Clarity type system includes the following types:
+
+- `(tuple (key-name-0 key-type-0) (key-name-1 key-type-1) ...)` - a typed tuple with named fields.
+- `(list max-len entry-type)` - a list of maximum length max-len, with entries of type entry-type
+- `(response ok-type err-type)` - object used by public functions to commit their changes or abort. May be returned or used by other functions as well, however, only public functions have the commit/abort behavior.
+- `(optional some-type)` - an option type for objects that can either be (some value) or none
+- `(buff max-len)` - byte buffer or maximum length max-len.
+- `(string-ascii max-len)` - ASCII string of maximum length max-len
+- `(string-utf8 max-len)` - UTF-8 string of maximum length max-len
+- `principal` - object representing a principal (whether a contract principal or standard principal).
+- `bool` - boolean value (true or false)
+- `int` - signed 128-bit integer
+- `uint` - unsigned 128-bit integer
+
+## Constructing Clarity Values
+Clarity values can be constructed with functions provided by the **@stacks/transactions** library. These function simply output javascript objects that contain a value and a numerical representation of the Clarity type information (or in the case of Booleans and Optional None, just the type information). These Clarity value objects can then be easily serialized into `contract-call` transactions for interacting with Clarity contracts.
+
+Here are examples of how to construct each type of Clarity value:
+
+### Booleans
+```javascript
+const t = trueCV();
+const f = falseCV();
+```
+### Optional Values
+```javascript
+const nothing = noneCV();
+const something = someCV(t);
+```
+
+### Buffers
+```javascript
+const buffer = Buffer.from('foo');
+const bufCV = bufferCV(buffer);
+```
+
+### Integers
+```javascript
+const i = intCV(-10);
+const u = uintCV(10);
+```
+
+### Strings
+```javascript
+const ascii = stringAsciiCV('hello world');
+const utf8 = stringUtf8CV('hello 🌾');
+```
+
+### Principals
+```javascript
+const address = 'SP2JXKMSH007NPYAQHKJPQMAQYAD90NQGTVJVQ02B';
+const contractName = 'contract-name';
+
+const spCV = standardPrincipalCV(address);
+const cpCV = contractPrincipalCV(address, contractName);
+```
+
+### Response Values
+```javascript
+const errCV = responseErrorCV(trueCV());
+const okCV = responseOkCV(falseCV());
+```
+
+### Tuples
+```javascript
+const tupCV = tupleCV({
+  'a': intCV(1),
+  'b': trueCV(),
+  'c': falseCV()
+})
+```
+
+### Lists
+```javascript
+const l = listCV([trueCV(), falseCV()])
+```
+
+## Using Clarity Values
+Now that you know how to construct Clarity values, you can use them to build `contract-call` transactions that call smart contract functions.
+
+Here's an example of how this is done:
```

Comment:
To avoid duplicated content, maybe we should just link to this chapter: https://docs.blockstack.org/stacks-blockchain/transactions#construction ?